SECURITY.md

```diff
@@ -285,6 +285,74 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<RespT> c
+
+If you have both `io.grpc:grpc-netty` and `io.grpc:grpc-okhttp`, you may also have issues. Remove `grpc-okhttp` if you are not on Android.
+
+If you have both `netty-codec-http2` and `netty-tcnative-boringssl-static` dependencies, then check the versions carefully. These versions could've been overridden by dependency management from another BOM. You would receive the "ALPN is not configured properly" exception if there are using incompatible versions.
+
+If you have other `netty` dependencies, such as `netty-all`, that are pulled in from other libraries, then ultimately you should make sure only one `netty` dependency is used to avoid classpath conflict. The easiest way is to exclude transitive Netty dependencies from all the immediate dependencies, e.g., in Maven use `<exclusions>`, and then add an explict Netty dependency in your project along with the corresponding `tcnative` versions. See the versions table below.
+
+If you are using `musl` libc (e.g., with Alpine Linux), then `netty-tcnative-boringssl-static` won't work either. There are several alternatives:
+ - Use [netty-tcnative-alpine] (https://github.com/pires/netty-tcnative-alpine)
+ - Use a distribution w/ `glibc`
+
+If you are running inside of an embedded Tomcat runtime (e.g., Spring Boot), then some versions of `netty-tcnative-boringssl-static` will have conflicts and won't work.
+
+Below are known to work version combinations:
+
+grpc-netty version | netty-code-http2 version | netty-tcnative-boringssl-static version | Conflicts w/ Embedded Tomcat
+------------------ | ------------------------ | --------------------------------------- | ----------------------------
+1.0.0              | 4.1.3.Final              | 1.1.33.Fork19                           | Yes
+1.1.0              | 4.1.8.Final              | 1.1.33.Fork26                           | Yes
+1.2.0              | 4.1.8.Final              | 1.1.33.Fork26                           | Yes
+1.3.0              | 4.1.8.Final              | 1.1.33.Fork26                           | Yes
+1.4.0              | 4.1.11.Final             | 2.0.1.Final                             | No
+1.5.0              | 4.1.12.Final             | 2.0.5.Final                             | No
+1.6.1              | 4.1.14.Final             | 2.0.5.Final                             | No
+1.7.0              | 4.1.16.Final             | 2.0.6.Final                             | No
+
+### OkHttp
+If you are using gRPC on Android devices, you are most likely using `grpc-netty` transport.
+
+Find the dependency tree (e.g., `mvn dependency:tree`), and look for versions of:
+ - `io.grpc:grpc-okhttp`
+ - `com.squareup.okhttp:okhttp`
+
+If you don't have `grpc-okhttp`, you should add it as a dependency.
+
+If you have both `io.grpc:grpc-netty` and `io.grpc:grpc-okhttp`, you may also have issues. Remove `grpc-netty` if you are on Android.
+
+If you have `okhttp` version below 2.5.0, or, if you have okhttp 3.x, then it may not work with gRPC.
```

Comment:
All versions 2.5.0 and later should work fine with gRPC. OkHttp3 has a different group name and package name, so you'd simply have both OkHttp 2 and OkHttp 3 in your classpath. Version incompatibilities (other than downgrading) with okhttp have not been a problem since v0.12.0 (yes, in 2015).

Reply:
ack; updated verbiage